Silicon/TigerlakePkg/Library/ItbtPcieRpLib/ItbtPcieRpLib.c

```diff
@@ -0,0 +1,215 @@
+  @param[out] Device                Return corresponding root port device number.
+  @param[out] Function              Return corresponding root port function number.
+
+  @retval     EFI_SUCCESS           Root port Info is retrieved
+  @retval     EFI_UNSUPPORTED       PortIndex is invalid/disabled, or TcssHob is not existed when mItbtPcieRpInfoUpdated is FALSE.
+**/
+EFI_STATUS
+EFIAPI
+GetItbtPcieRpInfo (
+  IN   UINTN    PortIndex,
+  OUT  UINTN    *Segment,
+  OUT  UINTN    *Bus,
+  OUT  UINTN    *Device,
+  OUT  UINTN    *Function
+  )
+{
+  //TCSS_DATA_HOB            *TcssHob;
+  //UINT32                   PcieSegMap;
+  //UINT32                   PcieFuncMap;
+
+  //TcssHob   = NULL;
+  //PcieSegMap  = 0;
+  //PcieFuncMap = 0;
+
+  //
+  // Iniatialize the Segment, Bus, Device, Function
+  //
+  *Segment  = 0;
+  *Bus      = 0;
+  *Device   = 0;
+  *Function = 0;
+
+  if(PortIndex >= MAX_ITBT_PCIE_PORT) {
+    DEBUG ((DEBUG_ERROR, "Invalid PortIndex %x", PortIndex));
+    return EFI_UNSUPPORTED;
+  }
+
+  if (mItbtPcieRpInfoUpdated == TRUE) {
+    if (mIbtbPcieRpInfo[PortIndex].Enable == TRUE) {
+      *Segment  = mIbtbPcieRpInfo[PortIndex].Segment;
+      *Bus      = mIbtbPcieRpInfo[PortIndex].Bus;
+      *Device   = mIbtbPcieRpInfo[PortIndex].Device;
+      *Function = mIbtbPcieRpInfo[PortIndex].Function;
+      return EFI_SUCCESS;
+    }
+  } /*else { //SG-todo - figure out how to get the TCSS Data HOB
+    TcssHob = (TCSS_DATA_HOB *) GetFirstGuidHob (&gTcssHobGuid);
+    if (TcssHob != NULL) {
+      PcieSegMap = TcssHob->TcssData.ItbtPcieSegMap;
+      PcieFuncMap= TcssHob->TcssData.ItbtPcieFuncMap;
+      //
+      // Update the Segment/Function pointers based on PcieSegMap & PcieFuncMap when Port enabled
+      //
+      if (TcssHob->TcssData.ItbtPcieRpEn[PortIndex]) {
+        *Segment  = (PcieSegMap >> (4 * PortIndex) ) & 0xF;
+        *Bus      = mIbtbPcieRpInfo[PortIndex].Bus;
+        *Device   = mIbtbPcieRpInfo[PortIndex].Device;
+        *Function = (PcieFuncMap >> (3 * PortIndex) ) & 0x7;
+        return EFI_SUCCESS;
+      }
+    }
+  }*/
+
+  return EFI_UNSUPPORTED;
+}
+
+// /**
+  // Get ITBT PCIE Root Port Index by ITBT PCIE Root Port Segment Bus Device and Function Number
+
+  // @param[in]  Segment             Root port segment number.
+  // @param[in]  Bus                 Root port Bus.
+  // @param[in]  Device              Root port Device.
+  // @param[in]  Function            Root port Function.
+  // @param[out] PortIndex           Return corresponding Root Port index (0-based)
+
+  // @retval     EFI_SUCCESS           PortIndex is retrieved
+  // @retval     EFI_UNSUPPORTED       PortIndex is invalid/disabled or TcssHob is not existed when mItbtPcieRpInfoUpdated is FALSE.
+// **/
+// EFI_STATUS
+// EFIAPI
+// GetItbtPciePortIndex (
+  // IN   UINTN    Segment,
+  // IN   UINTN    Bus,
+  // IN   UINTN    Device,
+  // IN   UINTN    Function,
+  // OUT  UINTN    *PortIndex
+  // )

```

Comment:
need remove this function if it is not used.

Reply:
Noted, I will remove the function.